### ISSUE TITLE
odd tags with realscripts

### ISSUE THREAD

**Issue by mbertucci47** [latex3/tagging-project#457]
With the realscripts package, the resulting text with `\textsuperscript` and `\textsubscript` is not correct in the tags and is not read correctly by Acrobat's "Read out loud".
```tex
\DocumentMetadata
  {
    lang=en-US,
    pdfversion=2.0,
    pdfstandard=ua-2,
    testphase={phase-III,math,title,table,firstaid},
  }
\documentclass{article}
\usepackage{fontspec}
\setmainfont{Libertinus Serif}
\usepackage{realscripts}

\begin{document}

Text\textsuperscript{12345678910abcdef}

Text\textsubscript{12345678910abcdef}

% original behavior
Text\textsuperscript*{12345678910abcdef}

Text\textsubscript*{12345678910abcdef}

\end{document}
```
<img width="197" alt="realscripts_struc" src="https://github.com/user-attachments/assets/605af925-b40f-4b7f-bdf3-e52e4b726cb1">

Similar things happen with other fonts, though some fonts like Source Serif Pro seem to only have superscript letters.

**Comment by FrankMittelbach** [comment]
What's your take @u-fischer ?

**Comment by FrankMittelbach** [comment]
Looking at that again, it seems to me this is a problem of Acrobat not of the file itself. Libertinus Serif seems to have all the sub and super script characters, but Acrobat uses a different font to display the tags and that one only as a smaller set of superscript chars, ie only the digits. In my version of Acrobat it seems to use yet another font so I get this:

<img width="266" alt="Screenshot 2024-11-14 at 23 01 45" src="https://github.com/user-attachments/assets/9a4c97f2-8c93-4078-8eb7-d4d075616646">

The read-aload in Acrobat is indeed funny:   "Text raised to the power of 2 to the power of 3" for the first line and nothing else. Seems to me that this is pure heuristics and the heuristic is unable to do anything with other Unicode chars in raised position.

So bottom line this seems to be compatible as far as it goes -- @u-fischer ?

But what may be considered wrong is the way the core commands in LaTeX work, ie, they just raise the normal chars but to not give them appropriate Unicode code points, however, that might be hard to fix if possible at all. Anyway, that is not a problem related to the package, so it should get its own issue.

**Comment by u-fischer** [comment]
Well yes, what you get with realscripts is font dependant (one reason why I don't like it if it loaded by default).  

I wouldn't worry to much about what is show in the tag tree, but I have doubt about the "appropriate" Unicode code points in case that means "non normal chars". Your own test show that it is not clear how a reader should read  the small superscript a (U+1D43 from the phonetic extension block) or the superscript four (U+2074).

Personally I think that in both cases it should be tagged as normal chars, so something like
`Span /TextPosition = Sup /ActualText =(123456789abcde)`

**Comment by FrankMittelbach** [comment]
> Personally I think that in both cases it should be tagged as normal chars, so something like
> `Span /TextPosition = Sup /ActualText =(123456789abcde)`

I agree, but then this is something to be done for the kernel `\textsuperscript` as well. So perhaps the answer is  to have a TaggingSocket with two arguments (first empty) that generated the span and the actual text if tagging is enabled. That could then be reused in the package as well.
